libs/api/utils/src/lib/error.ts

```diff
@@ -0,0 +1,46 @@
+import { isApiErrorCode } from '../api'
+import { type ApiErrorCode, type DocumentedErrorDescription, ERROR_LIST, type ErrorDefinition } from '../errorList'
+
+interface ApiError {
+  code?: number
+  message?: string
+  state?: string
+}
+
+export function parseError(error: Error | string | ApiError | unknown): DocumentedErrorDescription {
+  const e = normalizeType(error)
+  const message = (typeof e === 'string' ? e : e.message) || 'Internal Server Error'
+  const errorDefinition: ErrorDefinition | undefined = ERROR_LIST[message as ApiErrorCode]
+  const state = (() => {
+    if (!error) return ''
+    if (typeof error === 'object') {
+      if ('state' in error && typeof error.state === 'string') return error.state
+      if ('name' in error && typeof error.name === 'string') return error.name
+    }
+    return ''
+  })()
+
+  const errorCodeResolved: ApiErrorCode = isApiErrorCode(message) ? message : 'PUBLIC_COMMON_RAW_ERROR'
+
+  return {
+    state: (state || 'exception').toLowerCase(),
+    code: 500,
+    message: errorDefinition?.message || message,
+    errorCode: errorCodeResolved, // TODO: TS error. I dont know why.
```

Comment:
I do not understand how to fix this error.

<img width="2493" alt="Snímek obrazovky 2024-10-02 v 13 02 29" src="https://github.com/user-attachments/assets/59081594-3b08-4668-89c6-0fe41d840ba7">

Reply:
It is weird that it errors on this line, because true error is on line 33, where `message` is string. I'll check it out.

Reply:
It is weird that it errors on this line, because true error is on line 33, where message is string. I'll check it out.

Reply:
Fixed. Thank you.